ckanext/spectrum/tests/test_authn.py

```diff
@@ -28,6 +28,27 @@ def test_error_raised_for_non_sysadmin_user(self, app):
             }
         }
 
+    @pytest.mark.parametrize('action', [
+        ('user_show'),
+        ('package_list'),
+        ('package_search'),
+        ('user_list'),
+        ('dataset_duplicate'),
+        ('user_create')
+    ])
+    def test_api_endpoints_require_registered_user(self, app, action):
+        response = app.get(
+            toolkit.url_for('api.action', ver=3, logic_function=action),
```

Comment:
`toolkit.url_for('api')`

Reply:
Unfortunately url_for throws an error unless everything is given for a valid url route. This doesn't work.